baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/controller/AdminResource.java

```diff
@@ -0,0 +1,101 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.baaas.dfm.app.controller;
+
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.enums.SecuritySchemeType;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityScheme;
+import org.eclipse.microprofile.openapi.annotations.security.SecuritySchemes;
+import org.kie.baaas.dfm.app.controller.modelmappers.DecisionMapper;
+import org.kie.baaas.dfm.app.manager.DecisionLifecycle;
+import org.kie.baaas.dfm.app.manager.DecisionLifecycleOrchestrator;
+import org.kie.baaas.dfm.app.model.Decision;
+import org.kie.baaas.dfm.app.model.DecisionVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/admin")
+@SecuritySchemes(value = {
+        @SecurityScheme(securitySchemeName = "bearer",
+                type = SecuritySchemeType.HTTP,
+                scheme = "Bearer")
+})
+@SecurityRequirement(name = "bearer")
+public class AdminResource {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(AdminResource.class);
+
+    private final DecisionLifecycle decisionLifecycle;
+
+    private final DecisionMapper decisionMapper;
+
+    @Inject
+    public AdminResource(DecisionLifecycleOrchestrator decisionLifecycle,
+            DecisionMapper decisionMapper) {
+        requireNonNull(decisionLifecycle, "decisionLifecycle cannot be null");
+        requireNonNull(decisionMapper, "decisionMapper cannot be null");
+
+        this.decisionLifecycle = decisionLifecycle;
+        this.decisionMapper = decisionMapper;
+    }
+
+    @GET
+    @Path("/decisions")
+    @RolesAllowed({ "admin" })
+    public Response getDecisions() {
+        List<Decision> decisions = decisionLifecycle.listDecisions();
+        return Response.ok(decisions.stream()
+                .map(decisionMapper::mapToDecisionResponse)
+                .collect(toList())).build();
+    }
+
+    @DELETE
+    @Path("/decisions/{id}")
+    @RolesAllowed({ "admin" })
+    public Response deleteDecision(@PathParam("id") String id) {
+        LOGGER.info("Deleting decision with id or name '{}'...", id);
+        decisionLifecycle.deleteDecision(id);
+        return Response.ok().build();
+    }
+
+    @DELETE
+    @Path("/decisions/{id}/versions/{version}")
```

Comment:
What's the use case for deleting a specific version of a Decision?

Reply:
Hey @robpblake , the use case is that the user might deploy `n` versions of the same decision and only some of them are expired.

Reply:
So at what point is a Decision deleted entirely by our clean-up cronjob?

Reply:
@robpblake see here https://github.com/kiegroup/baaas-decision-infra/pull/11/files#r675510220 : the decision is erased if the `CURRENT` decision has expired or if all the decision versions are marked as `DELETED`. See the comment I posted here for details

Reply:
OK thanks @r00ta 👍